src/player.cpp

```diff
@@ -249,24 +250,49 @@ void Player::Run(IPC& ipc)
 			if(type == "keepalive")
 				keepAliveTimer = SDL_GetTicks();
 
-			if(type == "load"){
+			if(type == "load" || type == "lfsc_load"){
 				samples.clear();
 
-				video = Video::Create(message, 
-					// error handler
-					[&](Video::Error error, const std::string& msg){
-						if(error < Video::EEof)
-							ipc.WriteMessage("error", msg);
-						else
-							ipc.WriteMessage(error == Video::EEof ? "eof" : "unloaded", msg);
-					},
-					// audio handler
-					[&](const Sample* buffer, int size){
-						SDL_LockAudio();
-						for(int i = 0; i < size; i++)
-							samples.push(buffer[i]);
-						SDL_UnlockAudio();
-					}, 48000, 2, quickViewPlayer ? 5 : 60);
+				StreamPtr s;
+
+				FlogD("trying to load: " << message << " (" << type << ")");
+
+				try {
+					video = 0;
+
+					if(type == "load"){
+						FileStreamPtr fs = FileStream::Create();
+						fs->Open(message, false);
+						s = fs;
+					}
+
+					else {
+						s = lfsc->Open(Tools::StrToWstr(message));
+						s->Seek(0, SEEK_SET);
```

Comment:
Not entirely sure what you're referring to, specifically, but "load" creates a FileStream that loads from disk and "lfsc_load" creates an IpcStream that loads from a pipe/lfsc.